reactioncommerce/reaction
# GraphQL: Implement selectFulfillmentOptionForGroup mutation

Resolves #4513  
Impact: **breaking**  
Type: **feature**

## Changes
- The `selectFulfillmentOptionForGroup` GraphQL mutation now works
- The `setShipmentMethod` Meteor method now passes through to the `selectFulfillmentOptionForGroup` mutation

## Breaking changes
- The last argument of the `setShipmentMethod` Meteor method now expects just the method ID rather than the whole method object. Core client code has been updated, but you should update any custom code that calls this method.
- Removed the `resetShipmentMethod` Meteor method

## Testing
1. Verify that selecting a shipment method in checkout in the Meteor app still works as before.
1. Verify that you can use the new GraphQL mutation to select a shipment method.

```graphql
mutation {
  selectFulfillmentOptionForGroup(input: {cartId: "cmVhY3Rpb24vY2FydDo1UXFTUEgzUHM1ampRejZDWQ==", cartToken: "fpRHgSW3c7mgo4rgakZwHs1LVJlBuEksgpkHIPNBS3-", fulfillmentGroupId: "cmVhY3Rpb24vZnVsZmlsbG1lbnRHcm91cDptcTV3YzVQUkNwNHJmcGNaSg==", fulfillmentMethodId: "cmVhY3Rpb24vZnVsZmlsbG1lbnRNZXRob2Q6REVXMlBxWnJQZkVOUzQ0OEE="}) {
    cart {
      checkout {
        fulfillmentGroups {
          availableFulfillmentOptions {
            price {
              displayAmount
            }
            handlingPrice {
              displayAmount
            }
            fulfillmentMethod {
              _id
              name
              displayName
            }
          }
          selectedFulfillmentOption {
            price {
              displayAmount
            }
            handlingPrice {
              displayAmount
            }
            fulfillmentMethod {
              _id
              name
              displayName
            }
          }
        }
      }
    }
  }
}
```
